node/tests/sync_test.go

```diff
@@ -37,17 +36,79 @@ func TestSyncLightWithBridge(t *testing.T) {
 
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
-	time.Sleep(1 * time.Second)
-	assert.False(t, bridge.HeaderServ.IsSyncing())
+	h, err := bridge.HeaderServ.GetByHeight(ctx, 20)
+	require.NoError(t, err)
+
+	var ch int64 = 20
+	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockByHeight(ctx, &ch).BlockID.Hash)
 
 	addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(sw.Network.Host(bridge.Host.ID())))
 	require.NoError(t, err)
 	light := sw.NewLightNode(node.WithTrustedPeer(addrs[0].String()))
 
-	require.NoError(t, sw.Network.LinkAll())
 	err = light.Start(ctx)
 	require.NoError(t, err)
 
-	time.Sleep(1 * time.Second)
-	assert.False(t, light.HeaderServ.IsSyncing())
+	h, err = light.HeaderServ.GetByHeight(ctx, 30)
+	require.NoError(t, err)
+
+	ch = 30
+	assert.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockByHeight(ctx, &ch).BlockID.Hash)
+}
+
+/*
+Test-Case: Light Node continues sync after abrupt stop/start
+Pre-Requisites:
+- CoreClient is started by swamp
+- CoreClient has generated 50 blocks
+Steps:
+1. Create a Bridge Node(BN)
+2. Start a BN
+3. Check BN is synced to height 20
+4. Create a Light Node(LN) with a trusted peer
+5. Start a LN with a defined connection to the BN
+6. Check LN is synced to height 30
+7. Stop LN
+8. Start LN
+9. Check LN is synced to height 40
+*/
+func TestSyncStartStopLightWithBridge(t *testing.T) {
+	sw := swamp.NewSwamp(t, swamp.DefaultComponents())
+	ctx := context.Background()
+
+	sw.WaitTillHeight(ctx, 50)
+	bridge := sw.NewBridgeNode()
+	state := bridge.CoreClient.IsRunning()
+	require.True(t, state)
+	err := bridge.Start(ctx)
+	require.NoError(t, err)
+
+	h, err := bridge.HeaderServ.GetByHeight(ctx, 20)
+	require.NoError(t, err)
+
+	var ch int64 = 20
+	require.EqualValues(t, h.Commit.BlockID.Hash, sw.GetCoreBlockByHeight(ctx, &ch).BlockID.Hash)
```

Comment:
Pls, hide this int ugliness inside the `GetCoreBlockByHeight` method. You can pass here an int and make a pointer to it inside the method. Pls fix this everywhere.

Reply:
Agreed. Changed the func naming to Hash, too